pkg/users/account-details.js

```diff
@@ -343,3 +334,113 @@ export function AccountDetails({ accounts, groups, shadow, current_user, user })
+    const [validated, setValidated] = useState('default');
+    const [primaryGroupName, setPrimaryGroupName] = useState();
+    const previousValue = useRef(null);
+
+    useEffect(() => {
+        console.info({ groups });
+        const usedGroups = groups.filter(group => group.userlist.includes(name));
+        const primaryGroup = groups.find(group => group.userlistPrimary.includes(name));
+        const _primaryGroupName = primaryGroup ? primaryGroup.name : undefined;
+        const _selected = usedGroups.map(group => group.name);
+        if (primaryGroup)
+            _selected.push(_primaryGroupName);
+
+        const items_diff = previousValue.current
+            ? _selected
+                    .filter(x => !previousValue.current.includes(x))
+                    .concat(previousValue.current.filter(x => !_selected.includes(x)))
+            : [];
+        const validated = items_diff.length ? 'warning' : 'default';
+        previousValue.current = _selected;
+
+        setValidated(validated);
+        setSelected(_selected);
+        setPrimaryGroupName(_primaryGroupName);
+    }, [groups, setSelected, name, previousValue, setValidated]);
+
+    const removeGroup = group => {
+        return cockpit.spawn(["/usr/sbin/usermod", "-r", "-G", group, name], { superuser: "require", err: "message" })
+                .then(() => {
+                    setSelected(selected.filter(item => item !== group));
+                    setIsOpenGroup(false);
+                }, error => {
+                    show_unexpected_error(error);
+                });
+    };
+
+    const addGroup = group => {
+        return cockpit.spawn(["/usr/sbin/usermod", "-a", "-G", group, name], { superuser: "require", err: "message" })
+                .then(() => {
+                    setSelected([...selected, group]);
+                    setIsOpenGroup(false);
+                }, error => {
+                    show_unexpected_error(error);
+                });
+    };
+
+    const onSelectGroup = (event, selection) => {
+        if (selected.includes(selection)) {
+            removeGroup(selection);
+        } else {
+            addGroup(selection);
+        }
+    };
+
+    const chipGroupComponent = () => {
+        return (
+            <LabelGroup numLabels={10}>
+                {(selected || []).map((currentLabel, index) => {
+                    const optional = currentLabel !== primaryGroupName && superuser.allowed ? { onClose: () => removeGroup(currentLabel) } : {};
+
+                    return (
+                        <Label key={currentLabel}
+                               color={groups.find(group => group.name === currentLabel).isAdmin ? "gold" : "cyan"}
+                               {...optional}
+                        >
+                            {currentLabel}
+                        </Label>
+                    );
+                })}
+            </LabelGroup>
+        );
+    };
+
+    return (
+        <>
```

Comment:
Why this element when it only contains `FormGroup`?

Reply:
Fixed.